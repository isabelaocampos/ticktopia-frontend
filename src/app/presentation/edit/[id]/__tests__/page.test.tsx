import { render, screen } from '@testing-library/react'
import CreatePresentationPage from '../page'

// Mock Next.js useRouter - this is needed because UpdatePresentationForm uses it internally
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
    replace: jest.fn(),
    prefetch: jest.fn(),
    back: jest.fn(),
    forward: jest.fn(),
    refresh: jest.fn(),
  })),
}))

// Mock the UpdatePresentationForm component
jest.mock('../../../../../features/presentation/components/UpdatePresentationForm', () => {
  return function MockUpdatePresentationForm() {
    return <form data-testid="update-presentation-form">Update Presentation Form</form>
  }
})

describe('CreatePresentationPage (edit/[id]/page)', () => {

  it('renders the UpdatePresentationForm component', () => {
    render(<CreatePresentationPage />)
    
    expect(screen.getByTestId('update-presentation-form')).toBeInTheDocument()
  })

  it('renders the container div with correct classes', () => {
    render(<CreatePresentationPage />)
    const container = screen.getByTestId('update-presentation-form').parentElement
    expect(container).toHaveClass('max-w-xl', 'mx-auto', 'mt-10', 'p-4', 'border', 'rounded', 'shadow', 'space-y-6')
  })

  
})