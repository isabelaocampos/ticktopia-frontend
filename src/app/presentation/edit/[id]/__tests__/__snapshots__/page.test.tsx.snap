// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePresentationPage (edit/[id]/page) matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-xl mx-auto mt-10 p-4 border rounded shadow space-y-6"
  >
    <form
      data-testid="update-presentation-form"
    >
      Update Presentation Form
    </form>
  </div>
</DocumentFragment>
`;
